### DISCUSSION TITLE
Session 6 - 36 Lambda layer

### DISCUSSION THREAD

**Discussion by edwardborges** [chgasparoto/curso-aws-com-terraform#129] [Q&A] [answered]
Professor Bom dia

Fiz as configurações de acordo com o vídeo 
porém ao rodar terraform apply ele ocorreu um erro do local-exec provisioner error
```
 Error: local-exec provisioner error
│ 
│   with null_resource.build_lambda_layers,
│   on lambda.tf line 6, in resource "null_resource" "build_lambda_layers":
│    6:   provisioner "local-exec" {
│ 
│ Error running command 'npm install --production && cd ../ && zip -9 -r --quiet joi.zip *': exit status 127. Output: 
│ added 6 packages, and audited 7 packages in 3s
│ 
│ found 0 vulnerabilities
│ /bin/sh: 1: zip: not found

```
[plan-lambda-layer.txt](https://github.com/chgasparoto/curso-aws-com-terraform/files/10228244/plan-lambda-layer.txt)

Segue o arquivo do plan dessa mudança 
ao gerar o apply da o erro informado acima

O que errei?

Pode ajudar?

**Comment by edwardborges** [comment, marked as answer]
Professor Bom dia [chgasparoto](https://github.com/chgasparoto)

Atualizando, continuei vendo o curso até aula 40 da sessão 6 ,porém ainda estou com o problema relatado no post anterior 

ele mostra o erro do prosioner error , no momento que uso o terraform apply 
![image](https://user-images.githubusercontent.com/65915794/208435743-62b8eb2c-e159-490e-9c2f-377ed16a4ba3.png)

Pode ajudar? não consigo prosseguir para entender a parte de testando APi
Pois não cria meus recursos na AWS devido o error 

Fico no seu aguardo,

Atenciosamente ,

**Reply by chgasparoto** [reply]
@edwardborges está dando erro no `zip`, parece que vc não tem ele instalado na sua máquina. Vc precisa instalá-lo ou usar outro software para zipar os arquivos e atualizar o comando.

Pra instalar no Ubuntu:
```shell
apt-get update
apt-get install zip unzip
```

**Reply by edwardborges** [reply]
Nossa!!! como não percebi isso!!
Valeu demais professor
